src/modules/csm/csm.py

```diff
@@ -191,28 +199,35 @@ def module(self) -> StakingModule:
     def module_validators_by_node_operators(self, blockstamp: BlockStamp) -> ValidatorsByNodeOperator:
         return self.w3.lido_validators.get_module_validators_by_node_operators(self.module.id, blockstamp)
 
-    def _collect_data(self, blockstamp: BlockStamp) -> bool:
+    def validate_state(self, blockstamp) -> None:
+        assert self.state
+        converter = self.converter(blockstamp)
+        l_ref_slot, r_ref_slot = self.current_frame_range(blockstamp)
+        l_epoch = EpochNumber(converter.get_epoch_by_slot(l_ref_slot) + 1)
+        r_epoch = converter.get_epoch_by_slot(r_ref_slot)
+        for epoch in self.state.processed_epochs:
```

Comment:
we should check the missing epochs before building the report

Reply:
Done.